inst/include/code.h

```diff
@@ -0,0 +1,8 @@
+#ifndef FROGGER_H
+#define FROGGER_H
+
+inline int add(int a, int b) {
```

Comment:
This `inline` here is important because this is a concrete definition. If you'd done a template function like

```
template <typenameT > 
T add(T a, T b) {
  return a + b;
}
```

you would not need it as templates are implicitly inline. See https://en.cppreference.com/w/cpp/language/inline

This is important in your case because without it, every time you include this header you get another full definition of the function which violates the one-definition rule.